polling_stations/apps/data_importers/management/commands/import_eoni.py

```diff
@@ -90,9 +90,10 @@ def handle(self, *args, **options):
             }
             self.stations_only = options.get("stations_only")
             self.pre_process_data(reprojected=options["reprojected"])
-            self.clear_old_data()
-            self.copy_data()
-            self.assign_uprn_to_councils()
+            with transaction.atomic():
+                self.clear_old_data()
+                self.copy_data()
+                self.assign_uprn_to_councils()
```

Comment:
This is dealing with a separate issue. This is the inconsistent state the db can end up in id way through the import.
But it seemed reasonable to sneak it in here.